applications/serial_lte_modem/src/slm_at_socket.c

```diff
@@ -1599,7 +1599,22 @@ static int handle_at_getaddrinfo(enum at_cmd_type cmd_type, const struct at_para
 		if (err) {
 			return err;
 		}
-		err = getaddrinfo(host, NULL, NULL, &result);
+		if (param_count > 2) {
+			/* DNS query with designated address family */
+			struct addrinfo hints = {
+				.ai_family = AF_UNSPEC
+			};
+			err = at_params_int_get(param_list, 2, &hints.ai_family);
+			if (err) {
+				return err;
+			}
+			if (hints.ai_family < 0  || hints.ai_family > AF_INET6) {
+				return -EINVAL;
+			}
+			err = getaddrinfo(host, NULL, &hints, &result);
+		} else {
+			err = getaddrinfo(host, NULL, NULL, &result);
```

Comment:
```suggestion
		} else if (param_count == 2) {
			err = getaddrinfo(host, NULL, NULL, &result);
		} else {
			err = -EINVAL;
```

Reply:
Need to return in last `else` to save an unnecessary URC